tests/unit/test_io_utils.py

```diff
@@ -44,6 +44,13 @@ def test_assert_path_is_not_empty(self, tmp_path):
                 path=empty_temp_file_path, description="Io Utils"
             ), "Provided a fake file path, function is fail"
 
+        if len(os.listdir(tmp_path)) == 0:
```

Comment:
Since this line is testing whether the function raises an error if the provided directory path is empty it might be better placed before line 38. Because line 38 the `empty.txt` file is created in the tmp folder.